customer_service/src/Repositories/CustomerRepository.php

```diff
@@ -0,0 +1,394 @@
+<?php
+namespace Customer\Repositories;
+
+use Customer\Models\CustomerModel;
+use Lib\DatabaseManager\DatabaseManager;
+use Lib\ErrorReporter\ErrorReporter;
+use Lib\ErrorReporter\ErrorNodeFactory;
+
+/** Application level representation for executing the
+ * customer model related functions in the
+ * Domain Driven Design*/
+class CustomerRepository
+{
+    private $customer;
+    private $customers;
+
+    public function generatePassword($email)
+    {
+        $pass2=explode("@", $email);
+        $checkLen=strlen($pass2[0]);
+        $passInit=rand(1, $checkLen>=3?$checkLen-3:$checkLen);
+        $pass1=substr($email, $passInit, $checkLen>=3?3:$checkLen);
+        $password=$pass1."&Ab".rand(0, 99);
+        return $password;
+    }
+    
+    public function registerCustomer(
+        $firstName,
+        $lastName,
+        $phone,
+        $email,
+        $dob,
+        $gender,
+        $country,
+        $deviceType,
+        $region
+    ) {
+        $errorNodeFactory = new ErrorNodeFactory();
+        if ($this->customer != null) {
+            $errorNode = $errorNodeFactory->createObjectError(
+                "Customer Object already exist: will not create new user ",
+                "Internal error occured. Please try again later."
+            );
+            ErrorReporter::addNode($errorNode);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\Lib\ErrorReporter\ErrorReporter' in method 'registerCustomer'.](https://app.codacy.com/gh/BuildForSDG/Team-064-Backend/pullRequest?prid=5616370)